### PR TITLE
Notion: new chunking

### PR DESCRIPTION
**Pull request by philipperolet**
Related [task](https://github.com/dust-tt/dust/issues/2458)

Notion pages are now parsed into sections with their natural nesting structure, + h1/h2 adding a nesting level (even if the block "below" are not technically nested in them)

### CODE REVIEW

File: connectors/src/connectors/notion/temporal/activities.ts
Comment (philipperolet) [inline connectors/src/connectors/notion/temporal/activities.ts:1559]: Did not put much thought into that since it will change with the renderSectionContentTitle refactor by @spolu

File: connectors/src/connectors/notion/temporal/activities.ts
Comment (spolu) [inline connectors/src/connectors/notion/temporal/activities.ts:1976]: should be null no?

File: connectors/src/connectors/notion/temporal/activities.ts
Comment (spolu) [inline connectors/src/connectors/notion/temporal/activities.ts:1701]: let's move that here: https://github.com/dust-tt/dust/blob/main/types/src/core/data_source.ts#L29

File: connectors/src/connectors/notion/temporal/activities.ts
Comment (spolu) [reply connectors/src/connectors/notion/temporal/activities.ts:1559]: Let's rebase this on https://github.com/dust-tt/dust/pull/3151 as soon as merged.

Aren't you adding them at the end of the document here?

I think we can either add them to the top level content or even have them as prefix for the whole document.... hard to say for sure.

**Review by spolu — COMMENTED** [review]
Looks great, left a few comments

File: connectors/src/connectors/notion/temporal/activities.ts
Comment (philipperolet) [reply connectors/src/connectors/notion/temporal/activities.ts:1559]: > Aren't you adding them at the end of the document here?
yep, just kept that here as reminder

IMHO will add them as a second prefix in addition to title, along with author & lastEditor, using renderPrefix

Comment (spolu) [reply connectors/src/connectors/notion/temporal/activities.ts:1559]: I think we don't want renderPrefix for these do we? this will inevitably be quite messy if we split in the middle of a somewhat structured set of data.

Maybe add a limited number as prefix and the rest as content beyond a count

Comment (spolu) [reply connectors/src/connectors/notion/temporal/activities.ts:1565]: Or just take 3 and overflow if you have more?

File: connectors/src/connectors/notion/temporal/activities.ts
Comment (philipperolet) [inline connectors/src/connectors/notion/temporal/activities.ts:1724]: @fontanierh sanity check: pageHasBody flag replaced by computing parsed content length, seems ok but wanted to check with you in case the flag had a specific importance

File: connectors/src/connectors/notion/temporal/activities.ts
Comment (philipperolet) [inline connectors/src/connectors/notion/temporal/activities.ts:2050]: @fontanierh sanity check: moved image replacement logic from full parsed page to block level, seems ok but wanted to check with you in case it had a specific importance

**Review by spolu — APPROVED** [review]
LGTM assuming good testing of course :)

**Pull request merged**
